### ISSUE TITLE
profit is wrong in some cases

### ISSUE THREAD

**Issue by Tigzy** [freqtrade/freqtrade#3984]
## Describe your environment

  * Operating system: ubuntu server
  * Python Version: 3.6.9
  * CCXT version: 1.34.59
  * Freqtrade Version: develop-fe588e57
  
## Describe the problem:

Some trades (not all, far from it), show a negative profit while it's clearly not the case. 

![image](https://user-images.githubusercontent.com/9531438/99935276-a5667e00-2d60-11eb-993d-52a05eea6535.png)

### Steps to reproduce:

Have no idea how to reproduce
  
### Relevant code exceptions or logs

  ```
2020-11-23 02:18:22,242 - freqtrade.freqtradebot - INFO - Executing Sell for ETH/USDT. Reason: trailing_stop_loss
2020-11-23 02:18:22,775 - freqtrade.wallets - INFO - Wallets synced.
2020-11-23 02:18:23,260 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=2020-11-23 01:15:54)
2020-11-23 02:18:23,747 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=2020-11-23 01:15:54) [sell]: 0.05030694 B$
2020-11-23 02:18:23,747 - freqtrade.persistence.models - INFO - Updating trade (id=69) ...
2020-11-23 02:18:23,747 - freqtrade.persistence.models - INFO - MARKET_SELL has been fulfilled for Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=2020-11-23 01:$
2020-11-23 02:18:23,748 - freqtrade.persistence.models - INFO - Marking Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=closed) as closed as the trade is fulfill$
2020-11-23 02:18:24,332 - freqtrade.wallets - INFO - Wallets synced.
2020-11-23 02:18:24,333 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'trade_id': 69, 'exchange': 'Binance', 'pair': 'ETH/USDT', 'gain': 'loss', 'limit': 563.2200022388$
2020-11-23 02:18:25,840 - freqtrade.wallets - INFO - Wallets synced.

  ```

**Comment by Tigzy** [comment]
The trade is confirmed by Binance: 

![image](https://user-images.githubusercontent.com/9531438/99936116-a4cee700-2d62-11eb-96da-a47450fa06d0.png)

**Comment by xmatthias** [comment]
Based on the symptoms (and the screenshot) - i suspect this is a duplicate of #3922.

Unfortunately, the KEY log message for this (`2020-11-23 02:18:23,747 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=2020-11-23 01:15:54) [sell]: 0.05030694 B$`) seems cut off - as this should include the fee currency, and the fee rate (which it doesn't - so maybe you're able to provide the remainder of this line).

**Comment by Tigzy** [comment]
Sorry, here's the full line (not very good with nano :D)
`2020-11-23 02:18:23,747 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=69, pair=ETH/USDT, amount=0.08933000, open_rate=559.36000000, open_since=2020-11-23 01:15:54) [sell]: 0.05030694 BNB - rate: 0.03058935`

**Comment by Tigzy** [comment]
I suspect this is a fill issue, since the Binance claims it took 2 orders to fill it entirely (1 has BNB fees, the other ETH)

**Comment by Tigzy** [comment]
Ok this is a fee issue, the bot thinks it's BNB fees while it's a USD. There's a ~30$ ratio issue here (BNB price today)

**Comment by Tigzy** [comment]
What happens here I think it the bot is taking the first order currency (BNB), then sums up both
(0.00000013 + 0.05030681 = 0.05030694)

**Comment by xmatthias** [comment]
Feel free to submit a PR (including tests and such) that will fix this if you think it's that easy to fix.

The theory of what's happening is rather clear, but so far, i'm yet to get a full API response that includes this problem - which makes recreating it very difficult, as the `fetch_order` call you can do for the order placed now will not include the response included right after the order creation (see the other issue for this).

with that, all we can do is guess / try to recreate this.

It's not unfixable, but it'll take some time.

**Comment by Tigzy** [comment]
Sure thing, understanding the root cause is a first step.
At least it's not impacting the strategy, only the results/statistics. That's not too critical.

**Comment by xmatthias** [comment]
And that also only in very rare cases - as I've only seen 2 cases in over 3 years with this problem.

**Comment by Tigzy** [comment]
I guess the problem is just BNB balance. If you have dust they'll be consumed first, and then the rest is taken on currency sold.

**Comment by xmatthias** [comment]
no, BNB will be taken first, even if you own dust ... 

that's why i don't understand why you can have part of the order using BNB, and part in USDT assuming you've still got BNB left in the account...

**Comment by Tigzy** [comment]
`no, BNB will be taken first, even if you own dust ..., unless you're out of BNB`
That's what I'm saying by "dust", not enough to pay the whole fees

**Comment by xmatthias** [comment]
Could you please run the following (in a python shell within the environment that has ccxt installed) and post the output?

``` python
import ccxt
ct = ccxt.binance({
        'apiKey': 'YOUR_API_KEY',
        'secret': 'YOUR_SECRET',
})
--ct.verbose = True
order = ct.fetch_order('<orderid>', 'ETH/USDT')
print(order)
```


You'll need your apikey/secret (don't post them here!) - as well as the order id.

You should be able to get the order id by looking at the database with the following query. 
Best use sqlitebrowser or sqlite3 to open the database

`select * from orders where ft_trade_id = 69`

The order-id you're looking at should be the one with the highest id ... the one that's "filled" on the sell side.

I suspect the response will be similar to what we got in #3922 - which was not helpful as fee was not included ... but it's worth a try nonetheless.

**Comment by Tigzy** [comment]
I have another one. 
![image](https://user-images.githubusercontent.com/9531438/99990571-42063b80-2db4-11eb-9906-6eb5985412c4.png)

**Comment by Tigzy** [comment]
I'll try to get the binance ID, but it's not really immediate. I'll see what I can do.

**Comment by xmatthias** [comment]
This should be fixed with #4057 .

Obviously, this will not fix this for existing trades, only prevent new occurances.
Unfortunately, it's hard to test 100% reliably as this is a random occurance (which never happened to me personally) - but i tried to create a testcase which covers this.

Let me know if you encounter it again after upgrading to a version that contains this fix (current develop, or 2020.12) - obviously with logs so we can check where it went wrong.
